### DISCUSSION TITLE
LiveReload is an absolute curse, it runs in production and is terrible DX

### DISCUSSION THREAD

**Discussion by dagda1** [remix-run/remix#7141] [Q&A] [answered]
For the most part I am a fan of remix apart from the horrendous `<LiveReolad />`.

I'm used to hmr in other environments but I have to remember to comment out the `<LiveReload />` element when I deploy or weird things happen, like I have a route that uses mdx to render to HTML but after I forgot to comment out the `<liveReload />` element, it started returning `application/json` for reasons I do not understand:

<img width="849" alt="json" src="https://github.com/remix-run/remix/assets/118328/fdd14cf4-44be-4089-bd98-c7c93e2df7c5">

After I commented out the element then it started returning HTML again with a content-type of `text/html`

Is there an alternative to `<LiveReload />` or can I stop it doing evil things in production without commenting it out?

**Comment by kiliman** [comment, marked as answer]
That's odd. `<LiveReload>` should only render in *development*. Are you sure your production environment has the correct `NODE_ENV` set?

https://github.com/remix-run/remix/blob/06f801d4ee89d58a7e0abdc23baa2a516478d38f/packages/remix-react/components.tsx#L1830-L1841

**Reply by dagda1** [reply]
@kiliman you make a good point, I found this in the [breaking-changes](https://arc.codes/docs/en/about/upgrade-guide#breaking-changes) section of the aws architect docs.

> Due to ongoing issues with unpredictable behavior with certain external libraries, Architect no longer makes use of the NODE_ENV environment variable, nor is it automatically added to deployed apps.

**Reply by dagda1** [reply]
@kiliman you were very quick in closing this, I logged this to the console in my staging server:

```ts
console.log(process.env.NODE_ENV);
```

and it logs 'production';

There is something wrong with `<LiveReload />` or the code surrounding it.

I get a whole pile of these errors when it is not commented out no matter what the `NODE_ENV` is.

> In the minified production build of React, we avoid sending down full error messages in order to reduce the number of bytes sent over the wire.

**Reply by kiliman** [reply]
First of all, I didn't close the issue.

<img width="293" alt="image" src="https://github.com/remix-run/remix/assets/47168/62bb1c45-e121-44cd-9269-fc3eb51690c5">

How are you building and deploying your app to staging/production?

`esbuild` will evaluate `NODE_ENV` during the *build* process, so it shouldn't matter what the AWS runtime variable is.
